### PR TITLE
Return error and inhibitor severity to distinguish it in UI

### PR DESCRIPTION
**Pull request by andywaltlova**
OAMG-10608

I adjusted the message formats based on [conversion format](https://github.com/oamg/convert2rhel-worker-scripts/blob/main/scripts/c2r_script.py#L342). 

I also decided to count **error and inhibitor together**, in an essence errors also inhibit upgrade, report distinguishes them to show difference between system state and leapp erors - imho this is unnecessary burden for UI and ads complexity for users reading the short report message.

- [x] Adjust script tasks
- [x] Adjust ansible tasks
- [x] Test it - tested on rhel 8 and 7 respectively

See https://github.com/RedHatInsights/tasks-frontend/pull/147 for corresponding changes in severity maps (once approved I will change the draft status, I just want to make sure we are fine with errors to be shown as inhibitors)

### CODE REVIEW

**Comment by Monnte** [comment]
In my opinion it is fine that erros will be shown as inhibitors, but we should then change the message `"Your system has %s error%s and %s inhibitor%s out of %s potential problem%s."` to just mention inhibitors. Or add new severity for error. @andywaltlova choice is on you :smile:

File: scripts/leapp_script.py [hunk 1]
```diff
@@ -345,10 +353,9 @@ def parse_results(output, reboot_required=False):
             )
         )
         if reboot_required:
-            problem_info = message.lstrip("Your system has ").rstrip(".")
             message = (
-                "System will be upgraded (%s). Rebooting system in 1 minute."
-                % problem_info
+                "System will be upgraded. Rebooting system in 1 minute. "
+                "After reboot check inventory to verify the system is registered with new RHEL major version."
```
Comment (Monnte) [inline scripts/leapp_script.py:358]: I like the change :+1

File: playbooks/leapp_preupgrade_ansible.yml [hunk 2]
```diff
@@ -182,12 +186,57 @@
...
+              Your system has {{ error_count + inhibitor_count }} inhibitor{{ 's' if inhibitor_count + error_count != 1 else '' }}
+              out of {{ total_problems_count }} potential problems."
+          when: inhibitor_count != 0 or error_count != 0
+
+
+        - name: Set message if no inhibitors and no errors present
+          set_fact:
+            message: >-
+              {{ 'No problems found. The system is ready for upgrade.' if  total_problems_count == 0
+              else 'The upgrade can proceed. However, there is one or more warnings about issues that might occur after the upgrade.'}}
```
Comment (Monnte) [inline playbooks/leapp_preupgrade_ansible.yml:231]: Can we add this also to rhel7 pre/upgrade script ? To match the messages on both tasks.

File: scripts/leapp_script.py (hunk 1 above)
Comment (andywaltlova) [reply scripts/leapp_script.py:358]: yep, the report should be still in expanded section if any problems found, but it doesn't have to be in message anymore

File: playbooks/leapp_preupgrade_ansible.yml (hunk 2 above)
Comment (andywaltlova) [reply playbooks/leapp_preupgrade_ansible.yml:230]: I tried to sync it in last commit, take a look please - I am confused about all the tasks myself, but it should be fine now. I kept the error+inhibitor = inhibitor approach for all of them

File: playbooks/leapp_preupgrade_script.yaml [hunk 3]
```diff
@@ -333,29 +333,44 @@
...
-                      inhibitor_count,
-                      "" if inhibitor_count == 1 else "s",
-                      len(report_entries),
-                      "" if len(report_entries) == 1 else "s",
+
+              if inhibitor_count == 0 and error_count == 0:
+                  if total_problems_count == 0:
+                      message = "No problems found. The system is ready for upgrade."
+                  else:
+                      message = "The upgrade can proceed. However, there is one or more warnings about issues that might occur after the upgrade."
```
Comment (Monnte) [inline playbooks/leapp_preupgrade_script.yaml:357]: Missing test for this line :smile:

File: playbooks/leapp_preupgrade_ansible.yml (hunk 2 above)
Comment (Monnte) [reply playbooks/leapp_preupgrade_ansible.yml:230]: Yes, looks good now. Thanks!

File: playbooks/leapp_preupgrade_script.yaml (hunk 3 above)
Comment (andywaltlova) [reply playbooks/leapp_preupgrade_script.yaml:359]: updated :)

**Review by Monnte — APPROVED** [review]

**Pull request merged**